design/migration-path.md

```diff
@@ -0,0 +1,92 @@
+# Proposal: Migration to Conan 2.0 general guidelines
+
+| **Status**        | **Proposed/Accepted/Deprecated**             |
+|:------------------|:---------------------------------------------|
+| **RFC #**         | ####                                         |
+| **Submitted**     | 2020-11-25                                   |
+| **Dependencies**  | RFC #, #                                     |
+
+---
+
+## Summary
+This document defines some general principles about the future upgrade to Conan 2.0 path that users will need to do.
+In summary:
+- Conan recipes will implement a subset of syntax and features in Conan 1.X that is fully operational and can run in Conan 2.0 without requiring changes in conanfiles.
+- The command line in Conan 2.0 will change, in a non 1.X backwards compatible way.
+- Behavior will change, in a non 1.X backwards compatible way.
+- Configuration files might change, in a non 1.X backwards compatible way, but trying to reduce the changes to a minimum.
+- When Conan 2.0 is released, it will become the active series, and Conan 1.X will only get important bug fixes.
+
+
+## Motivation
+Conan 2.0 is a great opportunity to get rid of some existing legacy, outdated or not optimal defaults, and deprecated practices
+and features. The only way to achieve this is by introducing some breaking behavior. On the other hand, it is necessary to
+define an upgrade path that is as smooth as possible, with a special focus on avoiding the split of the ecosystem like it happened
+with Python 2 and 3.
+
+
+## Proposal
+
+### Conan recipe compatible syntax 1.X => 2.0
+
+This doesn't mean that all Conan 1.X syntax and features will work in Conan 2.0. Many things will be deprecated and removed.
+It means that Conan will provide the features that will be defined as the canonical and recommended ones in Conan 2.0 in 1.X.
+These features will be declared stable in 1.X when there is enough usage evidence, to allow companies to safely adopt them.
```

Comment:
It's a bit unclear what does "Conan 1.X feature" mean: is it Conan 1.35+ (or whatever it will be at that moment) or Conan 1.0?
I quite often find that even Conan 1.x recipes are not compatible and just fixing the recipe using documentation for reference breaks it for some Conan 1.6 (which we've started with). Among examples are tuples vs lists (in settings or options) or using dicts for `default_options`.
And such old versions fail with obscure message like "Can not fetch dependencies" or so.

And I suspect many can still use such ancient versions (at least for some parts) because of build system rigidity or mistakes made during learning.

So my suggestion is to clarify what does "Conan 1.X" mean: Is it a most basic subset which works from 1.0 and onward or is it a specific version which users have to upgrade to, which supports both 1.0 syntax (with already some features deprecated) and subset of 2.0.

Like in python it was only 2.7 which was "source-compatible" with python 3.